prepare_stmt.go

```diff
@@ -50,23 +50,18 @@ func (db *PreparedStmtDB) prepare(ctx context.Context, conn ConnPool, isTransact
 	}
 	db.Mux.RUnlock()
 
-	db.Mux.Lock()
-	defer db.Mux.Unlock()
-
-	// double check
-	if stmt, ok := db.Stmts[query]; ok && (!stmt.Transaction || isTransaction) {
-		return stmt, nil
-	} else if ok {
-		go stmt.Close()
-	}
-
 	stmt, err := conn.PrepareContext(ctx, query)
```

Comment:
looks like need to `Prepare` after the lock?

Reply:
mean here?
https://github.com/go-gorm/gorm/pull/5568/files#diff-0b194a262799c7ce6ddec3469dedda37d182b3a97d16f079bc234778aaa6312eR60

Reply:
Yes, change it to something like:

```go
	db.Mux.Lock()
	stmt, err := conn.PrepareContext(ctx, query)
	if err != nil {
		return Stmt{}, err
	}

	cacheStmt := Stmt{Stmt: stmt, Transaction: isTransaction}
	db.Stmts[query] = cacheStmt
	db.PreparedSQL = append(db.PreparedSQL, query)
	db.Mux.Unlock()
```

Reply:
Okay I got it, we can't lock conn.PrepareContext, I'll fix the code later and describe why the deadlock happens.